activejob/lib/active_job/enqueue_after_transaction_commit.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module ActiveJob
+  module EnqueueAfterTransactionCommit # :nodoc:
+    extend ActiveSupport::Concern
+
+    included do
+      ##
+      # :singleton-method:
+      #
+      # Defines if enqueueing this job from inside an Active Record transaction
+      # automatically defers the enqueue to after the transaction commit.
+      #
+      # It can be set on a per job basis:
+      #  - `:always` forces the job to be deferred.
+      #  - `:never` forces the job to be queueed immediately
+      #  - `:auto` uses the queue adapter default behavior (recommended).
+      class_attribute :enqueue_after_transaction_commit, instance_accessor: false, instance_predicate: false, default: :never
+
+      around_enqueue do |job, block|
+        after_transaction = case job.class.enqueue_after_transaction_commit
+        when :always
+          true
+        when :never
+          false
+        else # :auto
+          queue_adapter.respond_to?(:enqueue_after_transaction_commit?) && queue_adapter.enqueue_after_transaction_commit?
+        end
```

Comment:
FYI, the alignment style that I've previously found to keep Rubocop happy is:

```suggestion
        after_transaction =
          case job.class.enqueue_after_transaction_commit
          when :always
            true
          when :never
            false
          else # :auto
            queue_adapter.respond_to?(:enqueue_after_transaction_commit?) && queue_adapter.enqueue_after_transaction_commit?
          end
```